### PR TITLE
EZP-31558: Updated loadByIdentifier to use escapeForCacheKey function before get value from cache

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31558](https://jira.ez.no/browse/EZP-31558)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.5`/`master`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 1]
```diff
@@ -239,9 +239,9 @@ static function (Type $type) use ($status, $getTypeKeysFn) {
     public function loadByIdentifier($identifier)
     {
         return $this->getCacheValue(
-            $identifier,
+            $this->escapeForCacheKey($identifier),
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:242]: Shouldn't `$this->escapeForCacheKey($identifier)` be moved higher to avoid calling it twice?

**Comment by mateuszdebinski** [comment]
> It would be good to verify, whether escaping is necessary in some other methods, e.g. `loadGroupByIdentifier` and handlers (if needed).

I will check and if necessary I make changes, for now, I change to WIP

**Comment by andrerom** [comment]
@mateuszdebinski You may go ahead an add this everywhere where we load by identifier or remote id I think, bette be safe then sorry.

File: eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php [hunk 2]
```diff
@@ -29,10 +29,10 @@ class ContentLanguageHandler extends AbstractInMemoryPersistenceHandler implemen
     protected function init(): void
     {
         $this->getTags = static function (Language $language) { return ['language-' . $language->id]; };
-        $this->getKeys = static function (Language $language) {
+        $this->getKeys = function (Language $language) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php:32]: Why removing `static`?

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 3]
```diff
@@ -43,10 +43,10 @@ class ContentTypeHandler extends AbstractInMemoryPersistenceHandler implements C
     protected function init(): void
     {
         $this->getGroupTags = static function (Type\Group $group) { return ['type-group-' . $group->id]; };
-        $this->getGroupKeys = static function (Type\Group $group) {
+        $this->getGroupKeys = function (Type\Group $group) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:46]: Why removing `static`?

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 4]
```diff
@@ -56,11 +56,11 @@ protected function init(): void
                 'type-' . $type->id,
             ];
         };
-        $this->getTypeKeys = static function (Type $type, int $status = Type::STATUS_DEFINED) {
+        $this->getTypeKeys = function (Type $type, int $status = Type::STATUS_DEFINED) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:59]: Why removing `static`?

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 5]
```diff
@@ -144,9 +144,9 @@ function (array $groupIds) {
     public function loadGroupByIdentifier($identifier)
     {
         return $this->getCacheValue(
-            $identifier,
+            $this->escapeForCacheKey($identifier),
             'ez-content-type-group-',
-            function (string $identifier) {
+            function () use ($identifier) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:149]: Is this change necessary?

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 6]
```diff
@@ -239,9 +239,9 @@ static function (Type $type) use ($status, $getTypeKeysFn) {
     public function loadByIdentifier($identifier)
     {
         return $this->getCacheValue(
-            $identifier,
+            $this->escapeForCacheKey($identifier),
             'ez-content-type-',
-            function ($identifier) {
+            function () use ($identifier) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:244]: Is this change necessary?

File: eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php [hunk 7]
```diff
@@ -103,9 +103,9 @@ function (array $ids) {
     public function loadByLanguageCode($languageCode)
     {
         return $this->getCacheValue(
-            $languageCode,
+            $this->escapeForCacheKey($languageCode),
             'ez-language-code-',
-            function ($languageCode) {
+            function () use ($languageCode) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php:108]: Is this change necessary?

File: eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php [hunk 8]
```diff
@@ -119,9 +119,9 @@ function ($languageCode) {
     public function loadListByLanguageCodes(array $languageCodes): iterable
     {
         return $this->getMultipleCacheValues(
-            $languageCodes,
+            array_map([$this, 'escapeForCacheKey'], $languageCodes),
             'ez-language-code-',
-            function (array $languageCodes) {
+            function () use ($languageCodes) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php:124]: Is this change necessary?

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php [hunk 9]
```diff
@@ -256,9 +256,9 @@ function ($identifier) {
     public function loadByRemoteId($remoteId)
     {
         return $this->getCacheValue(
-            $remoteId,
+            $this->escapeForCacheKey($remoteId),
             'ez-content-type-',
-            function ($remoteId) {
+            function () use ($remoteId) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:261]: Is this change necessary?

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php [hunk 10]
```diff
@@ -58,10 +58,10 @@ public function init(): void
         $this->getRoleTags = static function (Role $role) {
             return ['role-' . $role->id];
         };
-        $this->getRoleKeys = static function (Role $role) {
+        $this->getRoleKeys = function (Role $role) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/UserHandler.php:61]: Why removing `static`?

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php [hunk 11]
```diff
@@ -288,9 +288,9 @@ public function loadRoleByIdentifier($identifier, $status = Role::STATUS_DEFINED
         }
 
         return $this->getCacheValue(
-            $identifier,
+            $this->escapeForCacheKey($identifier),
             'ez-role-',
-            function ($identifier) {
+            function () use ($identifier) {
```
Comment (konradoboza) [inline eZ/Publish/Core/Persistence/Cache/UserHandler.php:293]: Is this change necessary?

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php (hunk 4 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:59]: to use the escapeForCacheKey function we need to change from a static function to normal

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php (hunk 3 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:46]: to use the escapeForCacheKey function we need to change from a static function to normal

File: eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php (hunk 2 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php:32]: to use the escapeForCacheKey function we need to change from a static function to normal

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php (hunk 10 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/UserHandler.php:61]: to use the escapeForCacheKey function we need to change from a static function to normal

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php (hunk 11 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/UserHandler.php:293]: without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php (hunk 9 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:261]: without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database

File: eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php (hunk 8 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php:124]: without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database

File: eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php (hunk 7 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentLanguageHandler.php:108]: without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php (hunk 6 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:244]: without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database

File: eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php (hunk 5 above)
Comment (mateuszdebinski) [reply eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php:149]: without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database

**Comment by lserwatka** [comment]
You can merge it up to ezplatform-kernel/1.0 and master
